lib/datasets.go

```diff
@@ -445,36 +449,59 @@ func (r *DatasetRequests) Save(p *SaveParams, res *reporef.DatasetRef) (err erro
 		return fmt.Errorf("option to make dataset private not yet implemented, refer to https://github.com/qri-io/qri/issues/291 for updates")
 	}
 
-	// From cmd/, an empty reference becomes "me/", but from api/, it becomes "" (empty string).
-	// We must allow empty references for the case when the --new flag is being used, since it
-	// can be used to generate a name from a bodypath.
-	// TODO(dlong): Fix me! Check for these cases, return reasonable errors, test at lib/ level.
-	ref, err := repo.ParseDatasetRef(p.Ref)
-	if err != nil && err != repo.ErrEmptyRef {
+	ref, err := dsref.ParseHumanFriendly(p.Ref)
+	if err == dsref.ErrBadCaseName {
+		// If dataset name is using bad-case characters, and is not yet in use, fail with error.
+		if !r.nameIsInUse(ref) {
+			return err
+		}
+		// If dataset name already exists, just log a warning and then continue.
+		log.Error(dsref.ErrBadCaseShouldRename)
+	} else if err == dsref.ErrEmptyRef {
+		// Okay if reference is empty. Later code will try to infer the name from other parameters.
+	} else if err != nil {
+		// If some other error happened, return that error.
 		return err
 	}
 
+	// Validate that username is our own, it's not valid to try to save a dataset with someone
+	// else's username. Without this check, base will replace the username with our own regardless,
+	// it's better to have an error to display, rather than silently ignore it.
+	pro, err := r.node.Repo.Profile()
+	if err != nil {
+		return err
+	}
+	if ref.Username != "" && ref.Username != "me" && ref.Username != pro.Peername {
+		return fmt.Errorf("cannot save using a different username than \"%s\"", pro.Peername)
+	}
```

Comment:
this seems like valuable logic to put elsewhere at some point. Not totally sure where yet. Off the top of my head it'd be a method on a `Profile`:

```go
profile.OwnsRef(ref dsref.Ref) bool
```